src/twisted/test/test_defgen.py

```diff
@@ -96,100 +146,32 @@ def testStackUsage(self):
         Make sure we don't blow the stack when yielding immediately
         available deferreds.
         """
-        return self._genStackUsage().addCallback(self.assertEqual, 0)
+
+        @deprecatedDeferredGenerator
+        def _genStackUsage():
+            for x in range(5000):
+                # Test with yielding a deferred
+                x = waitForDeferred(defer.succeed(1))
+                yield x
+                x = x.getResult()
+            yield 0
+
+        return _genStackUsage().addCallback(self.assertEqual, 0)
 
     def testStackUsage2(self):
         """
         Make sure we don't blow the stack when yielding immediately
         available values.
         """
-        return self._genStackUsage2().addCallback(self.assertEqual, 0)
-
-
-def deprecatedDeferredGenerator(f):
-    """
-    Calls L{deferredGenerator} while suppressing the deprecation warning.
-
-    @param f: Function to call
-    @return: Return value of function.
-    """
-    return runWithWarningsSuppressed(
-        [
-            SUPPRESS(
-                message="twisted.internet.defer.deferredGenerator was " "deprecated"
```

Comment:
I see that this was deprecated in 2015 ... I think that is ok to just remove this code rather than keep maintaining it.

Reply:
Agreed.